middleware/fastcgi/fastcgi.go

```diff
@@ -34,12 +36,23 @@ type Handler struct {
 	Rules []Rule
 }
 
+func (h Handler) DoesFileExist(path string) bool {
+	file := h.Root + path
+	if _, err := os.Stat(file); err == nil {
+		return true
+	}
+	return false
+}
+
 // ServeHTTP satisfies the middleware.Handler interface.
 func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
 	servedFcgi := false
+	indexFile := "index.php"
```

Comment:
Yes, this is specifically for PHP, which is my use case. This code needs to be generalized before merging in (specifically these variables should be configurable parameters in the CaddyFile). I've been discussing how best to do this with @mholt and he will be taking a look. Thanks for helping me clarify.